### PR TITLE
DBZ-3952 Store schema for tables not in schema collection list

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-3952

### CODE REVIEW

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotSourceIT.java [hunk 1]
```diff
@@ -307,6 +326,53 @@ public void shouldCreateSnapshotSelectively() throws Exception {
...
+        try (
+                final MySqlTestConnection db = MySqlTestConnection.forTestDatabase(DATABASE.getDatabaseName());
+                final JdbcConnection connection = db.connect();
+                final Connection jdbc = connection.connection();
+                final Statement statement = jdbc.createStatement()) {
+            statement.executeUpdate("INSERT INTO customers VALUES (default,'John','Lazy','john.lazy@acme.com')");
+        }
+
+        Testing.Print.enable();
+        final SourceRecords streamingRecords = consumeRecordsByTopic(1);
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotSourceIT.java:373]: Let's assert the record in some way.

File: debezium-core/src/main/java/io/debezium/relational/RelationalTableFilters.java [hunk 2]
```diff
@@ -97,6 +102,10 @@ public TableFilter eligibleDataCollectionFilter() {
         return eligibleTableFilter;
     }
 
+    public Predicate<TableId> eligibleForSchemaDataCollectionFilter() {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalTableFilters.java:105]: L

File: debezium-core/src/main/java/io/debezium/relational/RelationalTableFilters.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
     private final TableFilter tableFilter;
     private final Predicate<String> databaseFilter;
     private final String excludeColumns;
+    private final Predicate<TableId> schemaSnapshotPredicate;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalTableFilters.java:28]:
```suggestion
    private final TableFilter schemaSnapshotFilter;
```

Can you also add a JavaDoc comment to the field?

**Review by gunnarmorling — APPROVED** [review]
The fix LGTM, some comments on details inline. Thx, @jpechane!

**Comment by jpechane** [comment]
@gunnarmorling All comments are addressed.

**Review by gunnarmorling — APPROVED** [review]
LGTM. Thanks, @jpechane!

**Comment by gunnarmorling** [comment]
Is the test failure related?

```
2022-02-16T18:08:41.2684521Z [ERROR] io.debezium.connector.mysql.ReadBinLogIT.shouldCaptureSingleWriteUpdateDeleteEvents  Time elapsed: 0.267 s  <<< FAILURE!
2022-02-16T18:08:41.2685815Z org.junit.ComparisonFailure: expected:<[1]> but was:<[0]>
2022-02-16T18:08:41.2686632Z 	at io.debezium.connector.mysql.ReadBinLogIT.recordedEventData(ReadBinLogIT.java:334)
2022-02-16T18:08:41.2687319Z 	at io.debezium.connector.mysql.ReadBinLogIT.shouldCaptureSingleWriteUpdateDeleteEvents(ReadBinLogIT.java:189)
```

If not, could you capture it in Jira?

**Comment by jpechane** [comment]
@gunnarmorling I don't see it locally. So far I downloaded the logs and restarted the job. If it passes I'll just log it as an intermittent failure.

**Comment by jpechane** [comment]
@gunnarmorling Jira logged

**Comment by gunnarmorling** [comment]
Thx!

**Pull request merged**